### DISCUSSION TITLE
Svelte 4 support

### DISCUSSION THREAD

**Discussion by bergstromdev** [TanStack/table#4944] [General]
Hi guys,
Thanks for all the great work you're putting in to tanstack table. It's great!

I'm using svelte-table and I'm trying to upgrade my Sveltekit app to use Svelte 4, but ran in to an issue with svelte-table.
It seems that svelte-table (8.9.3) has a peer dependency to `"svelte": "^3.49.0"` which stops me from doing that.

It would be great to see Svelte 4 support in svelte-table.

**Comment by itssumitrai** [comment]
Just stumbled into the same, any plans to upgrade to using Svelte 4 ? It should be a simple plug-in change

**Comment by nasheomirro** [comment]
Also using svelte 4 right now, I'm not sure whether to create an issue for this or not since there doesn't seem to be one yet.

**Comment by sukeshpabolu** [comment]
I want to upgrade to svelte 4 but this is a hurdle for me

**Comment by itssumitrai** [comment]
I have opened up a PR here: https://github.com/TanStack/table/pull/4996

**Comment by Rar9** [comment]
was updated yesterday. so this can be closed

**Comment by sukeshpabolu** [comment]
Hi, Has anyone started using tan stack table with typescript? I am getting this error? Argument of type '{ render: (props?: {} | undefined, { $$slots, context }?: { $$slots?: {} | undefined; context?: Map<any, any> | undefined; } | undefined) => { html: any; css: { code: string; map: any; }; head: string; }; $$render: (result: any, props: any, bindings: any, slots: any, context: any) => any; } | (new (options: any) => ...' is not assignable to parameter of type 'ConstructorOfATypedSvelteComponent | null | undefined'.

Possible causes:
- You use the instance type of a component where you should use the constructor type
- Type definitions are missing for this Svelte Component. If you are using Svelte 3.31+, use SvelteComponentTyped to add a definition:
  import type { SvelteComponentTyped } from "svelte";
  class ComponentName extends SvelteComponentTyped<{propertyName: string;}> {}
